test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -0,0 +1,1083 @@
+        self.assertTrue(rv)
+        # Add the configuration
+        self.load_config(self.cfg0)
+        # Now load the hook
+        a = {'content-type': 'application/x-python',
+             'content-encoding': 'default',
+             'input-file': hook_file}
+        rv = self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(hook_file)
+        time.sleep(2)
+        self.assertTrue(rv == 0)
+
+    def load_config(self, cfg):
+        cfg_file = os.path.join(os.sep, 'tmp', self.random_name())
+        with open(cfg_file, "w") as fd:
+            fd.write(cfg)
+        a = {'content-type': 'application/x-config',
+             'content-encoding': 'default',
+             'input-file': cfg_file}
+        rv = self.server.manager(MGR_CMD_IMPORT, HOOK, a, self.hook_name)
+        os.remove(cfg_file)
+        time.sleep(2)
+        self.assertTrue(rv == 0)
+
+    def set_vntype(self, typestring='myvntype'):
+        pbs_home = self.server.pbs_conf['PBS_HOME']
+        vntype_file = os.path.join(pbs_home, 'mom_priv', 'vntype')
+        with open(vntype_file, "w") as fd:
+            fd.write(typestring)
+
+    def remove_vntype(self):
+        pbs_home = self.server.pbs_conf['PBS_HOME']
+        vntype_file = os.path.join(pbs_home, 'mom_priv', 'vntype')
+        try:
+            os.remove(vntype_file)
+        except:
+            pass
+
+    def wait_and_remove(self, filename):
+        i = 0
+        while i < 10:
+            if not os.path.isfile(filename):
+                time.sleep(1)
+                i += 1
+                continue
+            try:
+                os.remove(filename)
+            except:
+                pass
+            break
+
+    def xtest_cgroup_vntype_excluded(self):
+        """
+        Test to verify that cgroups are not enforced on nodes
+        that have an exclude vntype file set
+
+        This test is disabled because it times out while
+        running with other tests, but runs fine by itself.
+        """
+        name = 'CGROUP8'
+        self.load_config(self.cfg1 % ("", '"no_cgroups"', "", ""))
+        self.set_vntype("no_cgroups")
+        a = {'Resource_List.select': '1:ncpus=1:mem=300mb',
+             'Job_Name': name}
+        j = Job(TEST_USER, attrs=a)
+        j.create_script(self.eatmem_job1)
+        jid = self.server.submit(j)
+        a = {'job_state': 'R'}
+        self.server.expect(JOB, a, jid)
+        self.server.status(JOB, ['Output_Path', 'Error_Path'], jid)
+        (o, e) = (j.attributes['Output_Path'], j.attributes['Error_Path'])
+        rv = self.mom.log_match(".*no_cgroups is in the excluded vnode" +
+                                " type list: \['no_cgroups'\].*",
+                                regexp=True,
+                                max_attempts=3,
+                                starttime=self.server.ctime)
+        self.remove_vntype()
+        self.wait_and_remove(o.split(':')[1])
+        self.assertTrue(rv)
```

Comment:
@hirenvadalia If that is the case, then how would one go about cleaning up temporary files after a test case has failed? In this case, we want to test for a specific condition before we cleanup any files.